src/Drupal/Commands/sql/SanitizeCommands.php

```diff
@@ -52,4 +72,79 @@ public function sanitize()
         // All sanitize operations defined in post-command hooks, including Drush
         // core sanitize routines. See \Drush\Commands\sql\SanitizePluginInterface.
     }
+
+    /**
+     * @hook option sql-sanitize
+     * @option whitelist-uids A comma delimited list of uids corresponding to the user accounts exempt from
+     *     sanitization.
+     * @option whitelist-mails
+     *   A comma delimited list of mails corresponding to the user accounts exempt from sanitization.
+     *   wildcard can be used to target all mail accounts on a domain.
+     */
+    public function options($options = ['whitelist-uids' => '', 'whitelist-mails' => ''])
+    {
+    }
+
+    /**
+     * Handles wildcard mail addresses and conversion of user mails to uids.
+     *
+     * @hook pre-command sql-sanitize
+     */
+    public function handleMilWhitelist(CommandData $commandData)
+    {
+        $input = $commandData->input();
+        $whitelist_mails = StringUtils::csvToArray($input->getOption('whitelist-mails'));
+        $whitelist_uids = StringUtils::csvToArray($input->getOption('whitelist-uids'));
+        $uids_mail_list = $this->uidsByMails($this->handleMailWildcard($whitelist_mails));
+        $input->setOption('whitelist-uids', implode(",", array_merge($whitelist_uids, $uids_mail_list)));
+    }
+
+    /**
+     * Helper function which returns user ids based on their mail addresses.
+     *
+     * @param array $mail_list
+     *
+     * @return array
+     */
+    private function uidsByMails($mail_list)
+    {
+        //print_r($mail_list);
+        if (empty($mail_list)) {
+            return [];
+        }
+        $conn = $this->getDatabase();
```

Comment:
This should use an entityQuery, not a raw DB query.

Reply:
Done